contracts/IdleTokenFungible.sol

```diff
@@ -0,0 +1,759 @@
+    _pause();
+  }
+
+  // unpause deposits
+  function unpause() external {
+    require(msg.sender == TL_MULTISIG || msg.sender == DL_MULTISIG || msg.sender == owner(), '6');
+    _unpause();
+  }
+
+  /**
+   * It allows owner to modify allAvailableTokens array in case of emergency
+   * ie if a bug on a interest bearing token is discovered and reset protocolWrappers
+   * associated with those tokens.
+   *
+   * @param protocolTokens : array of protocolTokens addresses (eg [cDAI, iDAI, ...])
+   * @param wrappers : array of wrapper addresses (eg [IdleCompound, IdleFulcrum, ...])
+   */
+  function setAllAvailableTokensAndWrappers(
+    address[] calldata protocolTokens,
+    address[] calldata wrappers
+  ) external onlyOwner {
+    require(protocolTokens.length == wrappers.length, "2");
+
+    address protToken;
+    for (uint256 i = 0; i < protocolTokens.length; i++) {
+      protToken = protocolTokens[i];
+      require(protToken != address(0) && wrappers[i] != address(0), "0");
+      protocolWrappers[protToken] = wrappers[i];
+    }
+
+    allAvailableTokens = protocolTokens;
+  }
+
+  /**
+   * It allows owner to set the IdleRebalancerV3_1 address
+   *
+   * @param _rebalancer : new IdleRebalancerV3_1 address
+   */
+  function setRebalancer(address _rebalancer)
+    external onlyOwner {
+      require((rebalancer = _rebalancer) != address(0), "0");
+  }
+
+  /**
+   * It allows owner to set the fee (1000 == 10% of gained interest)
+   *
+   * @param _fee : fee amount where 100000 is 100%, max settable is 20%
+   */
+  function setFee(uint256 _fee)
+    external onlyOwner {
+      // 100000 == 100% -> 10000 == 10%
+      require((fee = _fee) <= FULL_ALLOC / 5, "5");
+  }
+
+  /**
+   * It allows owner to set the fee address
+   *
+   * @param _feeAddress : fee address
+   */
+  function setFeeAddress(address _feeAddress)
+    external onlyOwner {
+      require((feeAddress = _feeAddress) != address(0), "0");
+  }
+
+  /**
+   * It allows owner to set the max unlent asset percentage (1000 == 1% of unlent asset max)
+   *
+   * @param _perc : max unlent perc where 100000 is 100%
+   */
+  function setMaxUnlentPerc(uint256 _perc)
+    external onlyOwner {
+      require((maxUnlentPerc = _perc) <= 100000, "5");
+  }
+
+  /**
+   * Used by Rebalancer to set the new allocations
+   *
+   * @param _allocations : array with allocations in percentages (100% => 100000)
+   */
+  function setAllocations(uint256[] calldata _allocations) external {
+    require(msg.sender == rebalancer || msg.sender == owner(), "6");
+    _setAllocations(_allocations);
+  }
+
+  /**
+   * Used by Rebalancer or in openRebalance to set the new allocations
+   *
+   * @param _allocations : array with allocations in percentages (100% => 100000)
+   */
+  function _setAllocations(uint256[] memory _allocations) internal {
+    require(_allocations.length == allAvailableTokens.length, "2");
+    uint256 total;
+    for (uint256 i = 0; i < _allocations.length; i++) {
```

Comment:
nits: we can remove overflow check from `i++`.

Reply:
I tried to keep minimal changes from the original code so I would leave this as is given also that is not a user facing method